chia/consensus/multiprocess_validation.py

```diff
@@ -221,10 +222,13 @@ async def pre_validate_blocks_multiprocessing(
     diff_ssis: List[Tuple[uint64, uint64]] = []
     for block in blocks:
         if block.height != 0:
-            assert block_records.contains_block(block.prev_header_hash)
             if prev_b is None:
-                prev_b = block_records.block_record(block.prev_header_hash)
+                prev_b = await block_records.get_block_record_from_db(block.prev_header_hash)
 
+        # get_next_sub_slot_iters_and_difficulty() requires prev_b.prev_hash to
+        # be in the block record cache.
+        if prev_b is not None and prev_b.height != 0 and not block_records.contains_block(prev_b.prev_hash):
+            return [PreValidationResult(uint16(Err.INVALID_PREV_BLOCK_HASH.value), None, None, False)]
         sub_slot_iters, difficulty = get_next_sub_slot_iters_and_difficulty(
```

Comment:
we use the cache here as well and we traverse to different points that i think this breaks, for instance we need  the previous previous block and we need the last transaction block before this block's signage point

Reply:
maybe it passes the tests by "accident". This is definitely the part of the patch that I'm the least confident in.
We have a test that ensures this function fail with this error under certain circumstances, but the new reorg tests also need to be able to validate blocks whose immediate history is not in the cache. It was tempting to make all of this logic fall back to the DB if a block isn't in the cache, but that breaks another, existing, test.